mempool/cat/store.go

```diff
@@ -38,13 +38,20 @@ func (s *store) set(wtx *wrappedTx) bool {
 func (s *store) get(txKey types.TxKey) *wrappedTx {
 	s.mtx.RLock()
 	defer s.mtx.RUnlock()
-	return s.txs[txKey]
+	tx := s.txs[txKey]
+	if tx != nil && tx.isReserved() {
+		return nil
+	}
+	return tx
 }
 
 func (s *store) has(txKey types.TxKey) bool {
 	s.mtx.RLock()
 	defer s.mtx.RUnlock()
-	_, has := s.txs[txKey]
+	tx, has := s.txs[txKey]
+	if has && tx.isReserved() {
+		return false
+	}
 	return has
 }
```

Comment:
wasn't sure about this one @cmwaters 

do we want to return true even if a tx is reserved? does this defeat the purpose of reserving in the first place?

Reply:
I think this is correct. We don't want to mark the node as having the transaction until we have verified it. We might want to revisit this solution however when we add the tx status endpoint. There could be several states:

- Processing
- Pending
- Proposed
- Committed
- Evicted
- Rejected